x-pack/plugins/endpoint/endpoint_app_types.ts

```diff
@@ -0,0 +1,36 @@
+/*
```

Comment:
Added a types file at the top level so that certain types can be shared between the FE and the BE. Also, I think we should try to keep all types in one file. When trying to determine the type of a variable, it's much easier when there is a single place where all type definitions live. If the file becomes too big and hard to manage, we could revisit and make adjustments. Thoughts @oatkiller @kevinlog @paul-tavares @parkiino?

Reply:
One place seems okay for now.  Makes sense to me to break it out when the need arises

Reply:
i've been using 1 file for all types Resolver. It's a great file to read if you want to understand everything that's being modeled by Resolver.  If a module creates a type for private use, I would consider leaving that in the module itself.

Having few types makes your system easy to understand I think.